packages/monaco-graphql-editor/src/editor/OperationsEditor.tsx

```diff
@@ -0,0 +1,187 @@
+  range?: monaco.IRange;
+  command?: monaco.languages.CompletionItem['command'];
+};
+
+const toCompletionItemKind = (kind: lsCIK): languages.CompletionItemKind => {
+  const CIK = languages.CompletionItemKind;
+  const map = {
+    [lsCIK.Text]: CIK.Text,
+    [lsCIK.Method]: CIK.Method,
+    [lsCIK.Function]: CIK.Function,
+    [lsCIK.Constructor]: CIK.Constructor,
+    [lsCIK.Field]: CIK.Field,
+    [lsCIK.Variable]: CIK.Variable,
+    [lsCIK.Class]: CIK.Class,
+    [lsCIK.Interface]: CIK.Interface,
+    [lsCIK.Module]: CIK.Module,
+    [lsCIK.Property]: CIK.Property,
+    [lsCIK.Unit]: CIK.Unit,
+    [lsCIK.Value]: CIK.Value,
+    [lsCIK.Enum]: CIK.Enum,
+    [lsCIK.Keyword]: CIK.Keyword,
+    [lsCIK.Snippet]: CIK.Snippet,
+    [lsCIK.Color]: CIK.Color,
+    [lsCIK.File]: CIK.File,
+    [lsCIK.Reference]: CIK.Reference,
+    [lsCIK.Folder]: CIK.Folder,
+    [lsCIK.EnumMember]: CIK.EnumMember,
+    [lsCIK.Constant]: CIK.Constant,
+    [lsCIK.Struct]: CIK.Struct,
+    [lsCIK.Event]: CIK.Event,
+    [lsCIK.Operator]: CIK.Operator,
+    [lsCIK.TypeParameter]: CIK.TypeParameter,
+  };
+  return map[kind] || CIK.Text;
+};
+
+const toCompletion = (
+  entry: GraphQLWorkerCompletionItem
+): monaco.languages.CompletionItem => {
+  return {
+    range: entry.range as monaco.IRange,
+    kind: toCompletionItemKind(entry.kind as lsCIK),
+    label: entry.label,
+    insertText: entry.insertText ?? (entry.label as string),
+    insertTextRules: entry.insertText
+      ? languages.CompletionItemInsertTextRule.InsertAsSnippet
+      : undefined,
+    sortText: entry.sortText,
+    filterText: entry.filterText,
+    documentation: entry.documentation,
+    detail: entry.detail,
+    command: entry.command,
+  };
+};
+
+class GraphQLWorker {
+  doComplete(
+    documentModel: monaco.editor.IReadOnlyModel,
+    position: monaco.Position,
+    schema: GraphQLSchema
+  ): GraphQLWorkerCompletionItem[] {
+    const document = documentModel.getValue();
+    if (!document) {
+      console.log('no document');
+      return [];
+    }
+    if (!schema) {
+      console.log('no schema');
+    }
+    const graphQLPosition = toGraphQLPosition(position);
+    const suggestions = getAutocompleteSuggestions(
+      schema,
+      document,
+      graphQLPosition
+    );
+    return suggestions.map((suggestion) => this.toCompletion(suggestion));
+  }
+
+  toCompletion(
+    entry: CompletionItem,
+    range?: IRange
+  ): GraphQLWorkerCompletionItem {
+    return {
+      label: entry.label,
+      insertText: entry.insertText,
+      insertTextFormat: entry.insertTextFormat,
+      sortText: entry.sortText,
+      filterText: entry.filterText,
+      documentation: entry.documentation,
+      detail: entry.detail,
+      range: range ? toMonacoRange(range) : undefined,
+      kind: entry.kind,
+      command: entry.command
+        ? { ...entry.command, id: entry.command.command }
+        : undefined,
+    };
+  }
+}
+
+export const OperationsEditor: FC<{
```

Comment:
```suggestion
export const OperationEditor: FC<{
```

Should we use singular instead of plural?

Reply:
I took name that was mentioned in Issue, also plural name says that editor can contain several named operations